src/foam/dao/index/AltFindPlan.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.dao.index;
+
+import foam.core.FObject;
+
+import foam.dao.Sink;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Not;
+import foam.mlang.predicate.Predicate;
+
+public class AltFindPlan implements FindPlan {
+  protected Object state_;
+  protected Plan bestPlan_ = new NotFoundPlan();
```

Comment:
Make this a FindPlan instead of a Plan, then you don't need to cast below.

Reply:
Ok, I will change that.